### ISSUE TITLE
Add metadata of deployment controlling pods

### ISSUE THREAD

**Issue by jsoriano** [elastic/beats#20991]
**Describe the enhancement:**

Currently Beats are adding metadata about the kubernetes resource controlling the monitored resource. In the case of pods created by deployments, the resource controlling the pod is actually a replica set, that is controlled by the deployment. Beats add the metadata about the replica set (e.g. the `kubernetes.replicaset.name` field), but not about the deployment.
Users are usually more interested in the deployment, than in the intermediary replica set, add this information to the added metadata.

I think this metadata is being added from here: https://github.com/elastic/beats/blob/44e281f0985648e5cbc2a9e509178060736dd2ee/libbeat/common/kubernetes/metadata/resource.go#L85

Not sure if there are other cases with intermediary objects like the case of deployments and replicasets, maybe a general option is to recursively get all owners.

**Describe a specific use case for the enhancement or feature:**

As a user I may want to be able to monitor pods per deployment creating them.

**Comment by nerddelphi** [comment]
It’ll be great!

**Comment by leosunmo** [comment]
This is a big deal for us as the replica set isn't very interesting. Looking at the docs for Filebeat (https://www.elastic.co/guide/en/beats/filebeat/current/exported-fields-kubernetes-processor.html) it looks like this is already supported, but it's not according to the code (and my current setup, which only contains `kubernetes.replicaset.name` and `kubernetes.statefulset.name`)

**Comment by leosunmo** [comment]
For anyone struggling, here's how I got around it for now!
```
        if [kubernetes][replicaset][name] {
          grok {
            id => "k8s-grok-add-deployment"
            match => {
              "[kubernetes][replicaset][name]" => "(?<kubernetes.deployment.name>[a-z0-9\-]+)(?:-[bcdfghjklmnpqrstvwxz2456789]{7,10})"
            }
            tag_on_failure => ["_k8s_deployment_fail"]
          }
        }
        ```

**Comment by ChrsMark** [comment]
👍🏼 @leosunmo  for user not having Logstash in the pipeline [script processor](https://www.elastic.co/guide/en/beats/metricbeat/current/processor-script.html) could do the trick.

Anyways it should be resolved by #23610 and I see that there is a recent (built 7 hours ago) snapshot of `7.12` which you could use to test it `docker.elastic.co/beats/metricbeat:7.12.0-SNAPSHOT`, but please have in mind that SNAPSHOTs are not official releases.
